server.go

```diff
@@ -345,6 +345,43 @@ func newServer(listenAddrs []string, chanDB *channeldb.DB, cc *chainControl,
 		},
 		ChannelPruneExpiry: time.Duration(time.Hour * 24 * 14),
 		GraphPruneInterval: time.Duration(time.Hour),
+		QueryBandwidth: func(edge *channeldb.ChannelEdgeInfo) lnwire.MilliSatoshi {
+			// First, we'll check to see if we're on either side of
+			// this edge. If we are, then we'll query the switch
+			// directly.
+			if bytes.Equal(edge.NodeKey1Bytes[:], selfNode.PubKeyBytes[:]) ||
+				bytes.Equal(edge.NodeKey2Bytes[:], selfNode.PubKeyBytes[:]) {
+
+				cid := lnwire.NewChanIDFromOutPoint(
+					&edge.ChannelPoint,
+				)
+				link, err := s.htlcSwitch.GetLink(cid)
+				if err != nil {
+					// If the link isn't online, then we'll
+					// report that it has zero bandwidth to
+					// the router.
+					return 0
+				}
+
+				// If the link is found within the switch, but
+				// it isn't yet eligible to forward any HTLCs,
+				// then we'll treat it as if it isn't online in
+				// the first place.
+				if !link.EligibleToForward() {
+					return 0
+				}
+
+				// Otherwise, we'll return the current best
+				// estimate for the available bandwidth for the
+				// link.
+				return link.Bandwidth()
+			}
+
+			// If this is a link on the graph detected from us,
+			// then we'll just thread through the capacity of the
+			// edge as we know it.
+			return lnwire.NewMSatFromSatoshis(edge.Capacity)
```

Comment:
Style nit: Could rewrite this to return early and avoid most of the indentation above.